internal/controllers/topology/cluster/current_state.go

```diff
@@ -232,3 +264,48 @@ func (r *Reconciler) getCurrentMachineDeploymentState(ctx context.Context, clust
 	}
 	return state, nil
 }
+
+// calculateCurrentRef calculates the ref to use to get an object based on the ref in ClusterClass
+// and the ref in the managed topology (Cluster, KCP, MD, ...).
+// If group and kind in both refs are the same we use the version from ClusterClass.
+// We have to do this so the topology controller can work on the exact version specified
+// in the ClusterClass by diffing current and desired state objects of the same version during reconcile.
+// If group or kind are different we use the ref from the managed topology. In this case
+// the group/kind was changed in the ClusterClass. We can't use the version from ClusterClass as
+// it might not match the old group/kind. But that's fine as a change in group/kind will end up in a diff and a
+// rollout anyway.
```

Comment:
```suggestion
// calculateCurrentRef updates the current Ref so it maches the apiVersion in ClusterClass;
// this is required so the topology controller can  diff current and desired state objects of the same version during reconcile.
// If group/kind was changed in the ClusterClass, then the current Ref is not modified because it will end up in a diff and a rollout anyway.
```
praise for explaining why; proposing a simplified version

Reply:
Thx for the suggestion. I took yours and modified it slightly to reflect that the handed over currentRef is not modified (we always intentionally return a new ref)